### ISSUE TITLE
Incorrect (unstable) compose web UI rendering on MacOS / iOS Safari

### ISSUE THREAD

**Issue by erikhuizinga** [JetBrains/compose-jb#2474]
I have a project, hosted here: https://erikhuizinga.github.io/ranstax/
Source code: https://github.com/erikhuizinga/ranstax

A UI issue happens on 
- Macbook Pro M1 on macOS 12.6.1 (21G217) with Safari 16.1 (17614.2.9.1.13, 17614)
- iPhone 12 Mini (I could find out the OS and browser versions if necessary, Chrome and Safari are both affected)

This issue is that the page should look like so:
<img width="1114" alt="Screenshot 2022-11-16 at 23 13 11" src="https://user-images.githubusercontent.com/19374736/202305806-fd9141d9-d784-42c5-923e-7727cf5f32b2.png">

But it looks like so if the window is resized (sometimes, sometimes not, it seems to be unstable UI behaviour!):
<img width="1114" alt="Screenshot 2022-11-16 at 23 08 24" src="https://user-images.githubusercontent.com/19374736/202305929-61aa1acd-0f95-4b6b-bc67-288e7151224e.png">

There are various unintended UI issues in the second image, the foremost being the buttons of the number grid being too large vertically.

Note that in both images the page source is open and the 'number button grid' element is selected. Note especially the CSS `height` property, which is (correctly) `283px`. However, in the (second) wrong image, the item isn't displayed at that height! Is this an issue in my code, or an issue in compose-jb for web? And if the issue is mine, how do I proceed to debug/fix it?

**Comment by eymar** [comment]
Your code looks fine.

I reproduced the issue on my Mac (intel) in Safari only (Chrome and Firefox don't show the issue). My Safari version is 15.6.1

It tried to replace `gridTemplateColumns("1fr 1fr 1fr")` by `gridTemplateColumns("repeat(3, 33%)")` and it behaves as expected. It seems to be an issue with `fr` in Safari (as far as I know the Chrome for iOS is just a branded Safari).

**Comment by erikhuizinga** [comment]
Thanks! Looking into applying the fix in my project soon.

**Comment by erikhuizinga** [comment]
Closing this as the issue seems to be related to Safari being stuck in the past (paraphrased from https://stackoverflow.com/a/44773109/3169029), though it isn't a Safari bug.

Here's my workaround based on some research after @eymar's suggestion and inspired by the Stackoverflow answer I just linked to: https://github.com/erikhuizinga/ranstax/commit/c7ffb5dece9110c0e72819a7ff84999f5e088fa4?w=1

The solution for me was to keep the grid template columns as `1fr 1fr 1fr` or `repeat(3, 1fr)`, but make the `height` of the grid explicit and wrap it in a `Div { }` so that assumes the correct height. Safari apparently needs that.

Do note that @eymar's suggestion to use 33% might work if your columns don't have gaps, but if they do, the gaps might cause an overflow, because they're added after summing the percentages (the percentages are based on the 100% width of the parent, so 33% * 3 columns means 99% width of the parent is available to the columns, leaving only 1% to the gap. In my app's case the gaps were fixed at `8px`, likely more than 1%, so an overflow occurred!).
